### PR TITLE
Archive all of the binaries in outerloop CI jobs.

### PR DESCRIPTION
**Pull request by mellinoe**
We aren't actually archiving the binaries in outerloop jobs. We only do this for innerloop jobs. I've copied over the logic to do that.

### CODE REVIEW

**Comment by mellinoe** [comment]
@dotnet-bot  Test CI please

**Comment by mellinoe** [comment]
@dotnet-bot Test CI please

**Comment by mellinoe** [comment]
@mmitche What's up with the error I got? I don't think my change should have affected that line.

```
ERROR: (netci.groovy, line 310) No signature of method: javaposse.jobdsl.dsl.helpers.step.StepContext.helix() is applicable for argument types: (org.codehaus.groovy.runtime.GStringImpl) values:
```

File: netci.groovy [hunk 1]
```diff
@@ -229,21 +230,27 @@ def targetGroupOsMap = ['netcoreapp': ['Windows 10', 'Windows 7', 'Windows_NT',
                         if (osName == 'Windows 10' || osName == 'Windows 7' || osName == 'Windows_NT') {
                             batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build.cmd -framework:${targetGroup} -${configurationGroup}")
                             batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build-tests.cmd -framework:${targetGroup} -${configurationGroup} -outerloop -- /p:IsCIBuild=true")
+                            batchFile("C:\\Packer\\Packer.exe .\\bin\\build.pack .\\bin")
```
Comment (mmitche) [inline netci.groovy]: @mellinoe Can you check the side of this directory?  The bin directories also contained obj in some cases, and were rather large.

Comment (mellinoe) [inline netci.groovy]: I'll match what the Unix jobs do here.

**Comment by mellinoe** [comment]
@dotnet-bot Test CI please

**Comment by mellinoe** [comment]
I'm doing a build to make sure the binaries are archived properly:

https://ci.dot.net/job/dotnet_corefx/job/master/job/GenPRTest/job/outerloop_netcoreapp_portablelinux_debug/
https://ci.dot.net/job/dotnet_corefx/job/master/job/GenPRTest/job/outerloop_netcoreapp_windows_nt_debug/

**Comment by mellinoe** [comment]
@dotnet-bot Test CI please

**Comment by mellinoe** [comment]
Cool -- looks like it is working. Merging now.

**Pull request merged**